shell/platform/darwin/macos/framework/Source/FlutterViewController.mm

```diff
@@ -646,6 +646,8 @@ - (void)dispatchMouseEvent:(NSEvent*)event phase:(FlutterPointerPhase)phase {
     flutterEvent.scroll_delta_x = -event.scrollingDeltaX * pixelsPerLine * scaleFactor;
     flutterEvent.scroll_delta_y = -event.scrollingDeltaY * pixelsPerLine * scaleFactor;
   }
+
+  [_keyboardManager syncModifiersIfNeeded:event.modifierFlags timestamp:event.timestamp];
```

Comment:
Is there a way to test `FlutterViewController` as a whole so that we can make sure mouse events trigger flag synchronization? Now you only test that the method of the responder works, which is, in fact, not that useful (since we might change this internal method in the future and we'll have to change the test.)